studygroups/tests/api/test_learning_circle_api.py

```diff
@@ -489,3 +500,94 @@ def test_get_learning_circles_by_weekday(self):
+        sg.save()
+        sg.refresh_from_db()
+        generate_all_meetings(sg)
+
+        sg = StudyGroup.objects.get(pk=2)
+        sg.start_date = datetime.date(2019,5,30)
+        sg.end_date = sg.start_date + datetime.timedelta(weeks=2)
+        sg.save()
+        sg.refresh_from_db()
+        generate_all_meetings(sg)
+
+        sg = StudyGroup.objects.get(pk=3)
+        sg.start_date = datetime.date(2019,5,1)
+        sg.end_date = sg.start_date + datetime.timedelta(weeks=2)
+        sg.save()
+        sg.refresh_from_db()
+        generate_all_meetings(sg)
+
+        sg = StudyGroup.objects.get(pk=4)
+        sg.start_date = datetime.date(2019,5,30)
+        sg.end_date = sg.start_date + datetime.timedelta(weeks=2)
+        sg.draft = True
+        sg.save()
+
+        c = Client()
+
+        # upcoming scope
+        resp = c.get('/api/learningcircles/?scope=upcoming&draft=true')
+        data = resp.json()
+        self.assertEqual(resp.status_code, 200)
+        self.assertEqual(data["count"], 2)
+        self.assertEqual(data["items"][0]["id"], 1)
+        self.assertEqual(data["items"][1]["id"], 4)
+
+        # current scope
+        resp = c.get('/api/learningcircles/?scope=current')
+        data = resp.json()
+        self.assertEqual(resp.status_code, 200)
+        self.assertEqual(resp.json()["count"], 1)
+        self.assertEqual(data["items"][0]["id"], 2)
+
+        # completed scope
+        resp = c.get('/api/learningcircles/?scope=completed')
+        data = resp.json()
+        self.assertEqual(resp.status_code, 200)
+        self.assertEqual(resp.json()["count"], 1)
+        self.assertEqual(data["items"][0]["id"], 3)
+
+
+    @freeze_time("2019-05-31")
+    def test_get_learning_circles_next_meeting(self):
+        sg = StudyGroup.objects.get(pk=1)
+        sg.start_date = datetime.date(2019,5,30)
+        sg.end_date = sg.start_date + datetime.timedelta(weeks=2)
+        sg.save()
+        sg.refresh_from_db()
+        generate_all_meetings(sg)
+
+        c = Client()
+
+        resp = c.get('/api/learningcircles/?scope=current')
+        data = resp.json()
+        self.assertEqual(resp.status_code, 200)
+        self.assertEqual(resp.json()["count"], 1)
+        self.assertEqual(data["items"][0]["next_meeting_date"], "2019-06-06")
+
+
+    @freeze_time("2019-05-31")
+    def test_get_learning_circles_last_meeting(self):
+        sg = StudyGroup.objects.get(pk=1)
+        sg.start_date = datetime.date(2019,5,1)
+        sg.end_date = sg.start_date + datetime.timedelta(weeks=2)
+        sg.save()
+        sg.refresh_from_db()
+        generate_all_meetings(sg)
+
+        c = Client()
+
+        resp = c.get('/api/learningcircles/?scope=completed')
+        data = resp.json()
+        self.assertEqual(resp.status_code, 200)
+        self.assertEqual(resp.json()["count"], 1)
+        self.assertEqual(data["items"][0]["last_meeting_date"], "2019-05-15")
+
```

Comment:
@s-kennedy could you please add a test to ensure we get the behavior we want from the API. IOW send a request with incrementally more text - "Lea", "lear", "learn", "learni" ... - and  make sure we get the results we're expecting.

Reply:
I actually started another branch for this issue, I hadn't made the PR yet though because it's not really giving the results i wanted. But i'll push it so you can see what I'm trying, maybe you have a better idea of how to make the query.